### ISSUE TITLE
grafana elasticsearch singlestat  

### ISSUE THREAD

**Issue by tonyC2016** [grafana/grafana#5602]
i use grafana2.6.0,and elasticsearch ,i used singlestat wish to get a field's Unique Count value  in a day,
but  it give me a wrong number ,the right total number  is 454906, and grafana count value  also is 454906 ,but so the es data is right ,and i use cardinality aggs at es ,with the json["precision_threshold": 20000] ,get a right number,19946 and  i checked grafana url ,there is no parameter precision_threshold, so  when its have a big data, the  Unique Count value cant to be accurate, in my scene,the value is must be  corret, so how can i  fixed  it ?

**Comment by tonyC2016** [comment]
this is my es query json :
{
    "aggs": {
        "partyId_count": {
            "cardinality": {
                "field": "driverParty.partyId",
                "precision_threshold": 20000
            }
        }
    },
    "query": {
        "filtered": {
            "filter": {
                "bool": {
                    "must": [
                        {
                            "range": {
                                "inputDate": {
                                    "gte": "1468684800000",
                                    "lte": "1468771199999"
                                }
                            }
                        }
                    ]
                }
            }
        }
    }
}

**Comment by tonyC2016** [comment]
the grafana request url is :
{
    "size": 0,
    "query": {
        "filtered": {
            "query": {
                "query_string": {
                    "analyze_wildcard": true,
                    "query": "*"
                }
            },
            "filter": {
                "bool": {
                    "must": [
                        {
                            "range": {
                                "inputDate": {
                                    "gte": "1468771200000",
                                    "lte": "1468857599999"
                                }
                            }
                        }
                    ]
                }
            }
        }
    },
    "aggs": {
        "2": {
            "date_histogram": {
                "interval": "1d",
                "field": "inputDate",
                "min_doc_count": 0,
                "extended_bounds": {
                    "min": "1468771200000",
                    "max": "1468857599999"
                }
            },
            "aggs": {
                "1": {
                    "cardinality": {
                        "field": "driverParty.partyId"
                    }
                }
            }
        }
    }
}

**Comment by tonyC2016** [comment]
@torkelo ,oh ,i see ,but if i wanna fix it  on grafana 2.6.0,can i replace some file which file that u had updated ,can it be work? coz  i need fix the problem very hurry

**Comment by torkelo** [comment]
No, you will have to back port the patch to 2.6.x branch and do new build

**Comment by tonyC2016** [comment]
@torkelo  does the version 3.1.0  grafana add the feature,

**Comment by torkelo** [comment]
@tonyC2016 no

**Comment by tonyC2016** [comment]
@torkelo  i git the source ,and find the file metric_agg.js  about line 73,there is have precision_threshold  is it the feature we discuss？code like :
 case 'cardinality': {
          var precision_threshold = $scope.agg.settings.precision_threshold || '';
          $scope.settingsLinkText = 'Precision threshold: ' + precision_threshold;
          break;
        }

**Comment by torkelo** [comment]
@tonyC2016 just use git to cherry pick the commit into the branch you want it in  (3.1.x), or us build from master http://grafana.org/download/builds.html

**Comment by tonyC2016** [comment]
@torkelo  sorry to bother again ,i try to build the code at local ,and i do it as the doc http://docs.grafana.org/project/building_from_source/,  i never have use go before ,and it shows as below: no buildable Go source files
 go get -d -v github.com/grafana/grafana
github.com/grafana/grafana (download)
package github.com/grafana/grafana
        imports github.com/grafana/grafana
        imports github.com/grafana/grafana: no buildable Go source files in /home/grafana1/src/github.com/grafana/grafana

**Comment by tonyC2016** [comment]
@torkelo  go is right
[root@sonar-test-host3 grafana2]# go version
go version go1.4.2 linux/amd64

**Comment by tonyC2016** [comment]
@torkelo 
i fixed it with
go get github.com/grafana/grafana 
cd $GOPATH/grafana/grafana
go get -v ...
